src/printer.js

```diff
@@ -2522,13 +2522,20 @@ function printJSXChildren(path, options, print, jsxWhitespace) {
 
         if (/\S/.test(value)) {
           // treat each line of text as its own entity
-          value.split(/(\n\s*)/).forEach(line => {
+          const lines = value.split(/(\n\s*)/);
+
+          // if there's a single line of text, we don't want to force a
```

Comment:
comment is slightly inaccurate, maybe "we don't want to allow the user to force a hardline"